scripts/scraper-ng/vfile-reporter-summary.js

```diff
@@ -0,0 +1,133 @@
+const chalk = require("chalk");
+const statistics = require("vfile-statistics");
+
+/**
+ * Generate a summary report of VFile messages.
+ *
+ * @param {(VFile|Array.<VFile>|Error)} files - `VFile`, `Array.<VFile>`, or `Error`
+ * @param {Object} options
+ * @returns A string
+ */
+function reporter(files) {
+  // Undefined or `null`
+  if (!files) {
+    return "";
+  }
+
+  // A single Error
+  if ("name" in files && "message" in files) {
+    // We might want to implement the full reporter API at some point, but not now.
+    throw Error("Not implemented");
+  }
+
+  // A single file
+  if (!("length" in files)) {
+    files = [files];
+  }
+
+  // Actually generate the summary
+  return summarize(files);
+}
+
+function summarize(files) {
+  const count = countRules(files);
+
+  const sortedRules = Object.entries(count)
+    .map(([ruleId, details]) => ({
+      ruleId,
+      ...details
+    }))
+    .sort((a, b) => a.count - b.count)
+    .reverse();
+
+  const ruleFields = sortedRules.map(formatRule);
+
+  const columns = ruleFields[0].length;
+  const widths = ruleFields.map(rule => rule.map(field => field.length));
+  const maxWidths = widths.reduce(
+    (prev, curr) => prev.map((number, index) => Math.max(number, curr[index])),
+    Array(columns).fill(0)
+  );
+
+  const paddedFields = ruleFields.map(rule =>
+    rule.map((field, columnNumber) => field.padEnd(maxWidths[columnNumber]))
+  );
+
+  const prologue = [""];
+  const body = paddedFields.map(field => field.join("  "));
+  const epilogue = ["", formatStats(files, body)];
+
+  return [].concat(prologue, body, epilogue).join("\n");
+}
+
+/**
+ * Count each message ruleId found in the array of vfiles.
+ *
+ * @param {Array.<VFile>} files
+ * @returns
```

Comment:
If we are going to have these doc blocks, I think they should be useful, and to be useful this ought to explain what the return value is.

Reply:
OK, I've tried to fill out the doc blocks a bit.